vulture_os/toolkit/api_parser/extrahop/extrahop.py

```diff
@@ -0,0 +1,205 @@
+                    self.session = None
+                    self.access_token = None
+            else:
+                raise ExtrahopAPIError(f"Error at request, unknown method : {method}")
+
+            if response.status_code == 200:
+                break
+            elif response.status_code == 204:
+                return []
+
+            if "The specified access token is invalid." in response.text:
+                # Force fetching a new token
+                logger.debug(f"[{__parser__}]:__execute_query: Force fetching a new token", extra={'frontend': str(self.frontend)})
+                self.session = None
+                self.access_token = None
+
+            logger.debug(f"[{__parser__}]:__execute_query: retry += 1 (waiting 10s)",
+                         extra={'frontend': str(self.frontend)})
+            retry += 1
+            self.evt_stop.wait(10.0)
+
+        # response.raise_for_status()
+        if response.status_code != 200:
+            logger.error(f"[{__parser__}]:__execute_query: Error at Extrahop API Call URL: {url} Code: {response.status_code} Content: {response.content}\n" \
+                f"Response : {response}", extra={'frontend': str(self.frontend)})
+            raise ExtrahopAPIError(
+                f"Error at Extrahop API Call URL: {url} Code: {response.status_code} Content: {response.content}\n" \
+                f"Response : {response}")
+
+        return response.json()
+
+    def _connect(self):
+        if self.session is None:
+            self.session = requests.Session()
+            self.session.headers.update(self.HEADERS)
+            self.session.headers.update({"Authorization": f"Bearer {self.get_access_token()}"})
+        return True
+
+    def test(self):
+        try:
+            if self.session is None:
+                self._connect()
+
+            since = timezone.now() - timedelta(hours=24)
+
+            response = self.get_logs(since)
+
+            return {
+                "status": True,
+                "data": response
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def get_logs(self, since):
+        if self.session is None:
+            self._connect()
+
+        url = f"{self.extrahop_host}/api/v1/detections/search"
+        payload = {
+            "create_time": int(since.timestamp() * 1000),
+        }
+        return self._execute_query("POST", url, payload)
+
+    @staticmethod
+    def format_log(log):
+        log['timestamp_iso'] = datetime.fromtimestamp(log['create_time']/1000, tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.%f")
+        return json.dumps(log)
+
+    def execute(self):
+        since = self.last_api_call or (timezone.now() - timedelta(days=30))
+        to = timezone.now()
```

Comment:
use the `to` in queries, and limit queries to 24h ranges max

Reply:
As previously indicated, we can't deal with this parameter.